src/analysis/normed_space/star/spectrum.lean

```diff
@@ -151,3 +154,53 @@ noncomputable instance : continuous_linear_map_class F ℂ A B :=
   .. alg_hom_class.linear_map_class }
 
 end star_alg_hom
+
+end
+
+namespace weak_dual
+
+open continuous_map complex
+open_locale complex_star_module
+
+variables {F A : Type*} [normed_ring A] [normed_algebra ℂ A] [nontrivial A] [complete_space A]
+  [star_ring A] [cstar_ring A] [star_module ℂ A] [hF : alg_hom_class F ℂ A ℂ]
+
+include hF
+
+/-- This instance is provided instead of `star_alg_hom_class` to avoid type class inference loops.
+See note [lower instance priority] -/
+@[priority 100]
+noncomputable instance : star_hom_class F A ℂ :=
+{ coe := λ φ, φ,
+  coe_injective' := fun_like.coe_injective',
+  map_star := λ φ a,
+  begin
+    suffices hsa : ∀ s : self_adjoint A, (φ s)⋆ = φ s,
+    { rw ←real_part_add_I_smul_imaginary_part a,
+      simp only [map_add, map_smul, star_add, star_smul, hsa, self_adjoint.star_coe_eq] },
+    { intros s,
+      have := alg_hom.apply_mem_spectrum φ (s : A),
+      rw self_adjoint.coe_re_map_spectrum s at this,
+      rcases this with ⟨⟨_, _⟩, _, heq⟩,
+      rw [←heq, is_R_or_C.star_def, is_R_or_C.conj_of_real] }
+  end }
+
+/-- This is not an instance to avoid type class inference loops. See
+`weak_dual.complex.star_hom_class`. -/
+noncomputable def _root_.alg_hom_class.star_alg_hom_class : star_alg_hom_class F ℂ A ℂ :=
+{ .. hF, .. weak_dual.complex.star_hom_class }
+
+omit hF
+
+variable (A)
+/-- The `gelfand_transform` as a `star_alg_hom` when the scalar field is `ℂ`. -/
+noncomputable def gelfand_star_transform : A →⋆ₐ[ℂ] C(character_space ℂ A, ℂ) :=
```

Comment:
```suggestion
/-- The `gelfand_transform` as a `star_alg_hom` when the scalar field is `ℂ`. -/
noncomputable def gelfand_star_transform : A →⋆ₐ[ℂ] C(character_space ℂ A, ℂ) :=
```
My understanding is that this will actually be a `star_alg_equiv` 99% of the time (right?). In that case, how about renaming this to `gelfand_star_transform'` or something like that and keep the "good" name for the equiv version?

Reply:
Now that you mention it, I think it may be possible to not define this version of `gelfand_star_transform` at all, and only define the equiv (once I have refactored `alg_equiv` and then defined `star_alg_equiv`). I will try on the other branch and report back.

Reply:
Yes, it basically works. There was only one place where I really needed the `gelfand_transform` to be star-preserving, and everywhere else I could get away with `weak_dual.star_hom_class`; in the one place that needed it in #16488 I just inlined the proof. So I have removed the definition of `gelfand_star_transform` from this PR so we can preserve the name for the equivalence.